lib/excon/utils.rb

```diff
@@ -12,7 +12,9 @@ module Utils
 
     def binary_encode(string)
       if FORCE_ENC && string.encoding != Encoding::ASCII_8BIT
-        string.force_encoding('BINARY')
+        string.dup.force_encoding('BINARY')
```

Comment:
Do you think we should wrap this in a check for frozen? I think it might be helpful to do so, just to prevent duplicating and using a lot of memory when there is no need to do so. Does that make sense? What do you think? I guess it would then look something like:
```
    if FORCE_ENC && string.encoding != Encoding::ASCII_8BIT
      if string.frozen?
        string.dup.force_encoding('BINARY')
      else
        string.force_encoding('BINARY')
      end
    else
      string
    end
```

Reply:
@geemus I didn't care about a memory footprint seriously. In my failed case, the argument of binary_encode() was a tiny string like `""`. It takes much memory If the huge string passed to this method surely as you said.

I'll update my patch. Thanks.

Reply:
That makes total sense, no worries. I think in most cases it probably wouldn't be too much of an issue, but I'd prefer to be cautious (especially when it is easy to do so, like this). Thanks!